client-v2/src/util/frontsUtils.ts

```diff
@@ -32,15 +32,18 @@ const getFrontCollections = (
 const combineCollectionWithConfig = (
   collectionConfig: CollectionConfig,
   collection: CollectionWithNestedArticles
-): CollectionWithNestedArticles =>
-  Object.assign({}, collection, {
+): CollectionWithNestedArticles => {
+  return Object.assign({}, collection, {
```

Comment:
just in case you didn't know and fancied being super-concise in future, this might be expressed more briefly with object spreading and parenthesised function bodies as => ({ ...collection, { <etc> } }) 😄

Reply:
I tried to solve this but got type mismatches when the CollectionConfig is merged with Collection so will leave as is for now.